### DISCUSSION TITLE
ValueError(f"{func} does not have a return type annotation") when implementing LightningCLI

### DISCUSSION THREAD

**Discussion by djtschke** [Lightning-AI/pytorch-lightning#19637] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
Dear All,

I am trying to implement the LightningCLI class, but it tells me my model does not have any Type annotation, although I added class function type return type annotation.

What am I doing wrong?

Thank you for your input!

LG

Max

**Comment by mauvilsa** [comment]
@djtschke it is difficult to say what you are doing wrong without seen the code you are running. Please provide a minimal reproduction example.

**Comment by djtschke** [comment, marked as answer]
Hi Mauvilsa,

thank you for your reply. While trying to come up with a minimal reproduction example I actually found the solution to my problem:
LightningCLI() does not take the instance of the model as a parameter but the class itself (as I guess instantiation is then done by the pipeline itself). To be more clear:

Wrong / Not Working:
```
from lightning.pytorch.cli import LightningCLI
from MyFancyModel import MyFancyModelClass
from MyFancyDataModule import MyFancyDataModuleClass

model = MyFancyModelClass()
datamodule = MyFancyDataModuleClass()

LightningCLI = LightningCLI(model, datamodule)
```

**Correct / Working**:
```
from lightning.pytorch.cli import LightningCLI
from MyFancyModel import MyFancyModelClass
from MyFancyDataModule import MyFancyDataModuleClass

LightningCLI = LightningCLI(MyFancyModelClass, MyFancyDataModuleClass)
```

Hope that also helps others!

Cheers,

Max
